### PR TITLE
feat(db-connection-ui): Show Preferred DBs

### PR DESCRIPTION
**Pull request by hughhhh**
### SUMMARY

Added starter Step 1 for preferred dbs render. User will now see the preferred DBs set in the superset_config file, and will be in a top section to make it easier for people to get to the connection page (step 2)

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

https://www.loom.com/share/ab848326e2884b19bd248303e50f4da0

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/config.py [hunk 1]
```diff
@@ -1075,10 +1075,10 @@ def CSV_TO_HIVE_UPLOAD_DIRECTORY_FUNC(
 # use the "engine_name" attribute of the corresponding DB engine spec
 # in `superset/db_engine_specs/`.
 PREFERRED_DATABASES: List[str] = [
```
Comment (hughhhh) [inline superset/config.py:1077]: will remove before merging

File: superset-frontend/src/views/CRUD/data/database/DatabaseModal/styles.ts [hunk 2]
```diff
@@ -340,5 +355,30 @@ export const EditHeaderSubtitle = styled.div`
 `;
 
 export const SelectDatabaseStyles = styled.div`
-  margin: ${({ theme }) => theme.gridUnit * 4}px;
+  margin: 0 16px;
```
Comment (AAfghahi) [inline superset-frontend/src/views/CRUD/data/database/DatabaseModal/styles.ts:358]: use gridUnit

File: superset-frontend/src/views/CRUD/data/database/DatabaseModal/index.tsx [hunk 3]
```diff
@@ -299,6 +301,73 @@ const DatabaseModal: FunctionComponent<DatabaseModalProps> = ({
     }
   };
 
+  const setDatabaseModel = option => {
```
Comment (eschutho) [inline superset-frontend/src/views/CRUD/data/database/DatabaseModal/index.tsx:304]: can the param be "engine" if it's just a string? Maybe a type on that would help, too.

File: superset-frontend/src/views/CRUD/data/database/DatabaseModal/index.tsx [hunk 4]
```diff
@@ -299,6 +301,73 @@ const DatabaseModal: FunctionComponent<DatabaseModalProps> = ({
...
+    setDB({
+      type: ActionType.dbSelected,
+      payload: {
+        configuration_method: isDynamic
+          ? CONFIGURATION_METHOD.DYNAMIC_FORM
+          : CONFIGURATION_METHOD.SQLALCHEMY_URI,
+        engine: option,
+      },
+    });
+    console.log('isDynamic', isDynamic);
```
Comment (eschutho) [inline superset-frontend/src/views/CRUD/data/database/DatabaseModal/index.tsx:317]: nit remove

File: superset-frontend/src/views/CRUD/data/database/DatabaseModal/index.tsx [hunk 5]
```diff
@@ -299,6 +301,73 @@ const DatabaseModal: FunctionComponent<DatabaseModalProps> = ({
...
+  };
+
+  const renderAvailableSelector = () => (
```
Comment (eschutho) [inline superset-frontend/src/views/CRUD/data/database/DatabaseModal/index.tsx:320]: can we move step one into a separate file?

File: superset-frontend/src/views/CRUD/data/database/DatabaseModal/index.tsx [hunk 6]
```diff
@@ -299,6 +301,73 @@ const DatabaseModal: FunctionComponent<DatabaseModalProps> = ({
...
+            onClick={() => setDatabaseModel(database.engine)}
+            buttonText={database.name}
+          />
+        ))}
+    </div>
+  );
+
+  const renderModalFooter = () =>
+    db // if db show back + connect
+      ? [
```
Comment (eschutho) [inline superset-frontend/src/views/CRUD/data/database/DatabaseModal/index.tsx:352]: db && (
<>

would that work instead?

**Comment by eschutho** [comment]
This is looking good @hughhhh!

**Pull request merged**